### PR TITLE
[VNext][Runtime] Add parent-side delegated skill stubs and invoke_delegated_skill

### LINKED ISSUE #183: [VNext][Runtime] Add parent-side delegated skill stubs and invoke_delegated_skill
Related to #181.

Depends on #108 and #175.
Supports #10.

## Summary

Expose delegated skills to the parent as lightweight callable capabilities and add a runtime-native `invoke_delegated_skill` path instead of treating delegated skills as normal full-body prompt injections.

## Why

Once a skill resolves as delegated, Alan still needs a parent-side behavior that preserves the point of delegation.

If delegated skills are injected like normal inline skills, the parent still absorbs:

- the full skill body
- the child executor's workflow guidance
- much of the context delegated execution is meant to avoid

Delegated skills therefore need a different parent-side surface:

- lightweight capability stub
- explicit invocation path
- runtime-owned child launch

## Scope

- add a parent-side delegated capability stub for selected delegated skills
- avoid injecting full delegated-skill bodies into the parent prompt by default
- introduce a dedicated virtual tool such as `invoke_delegated_skill`
- map delegated invocation onto package-local child-agent execution through `SpawnSpec`
- keep default child launch fresh-by-default with explicit handles only
- start with minimal default handles:
  - `workspace`
  - `result_channel`

## Non-Goals

- do not add `delegate_preferred` fallback behavior in the first version
- do not bind parent tape, active skills, plan state, memory, or approval cache by default
- do not enable nested delegated execution in the first version

## Acceptance Criteria

- delegated skills appear to the parent as lightweight callable capabilities
- delegated skills are no longer injected as full inline prompt bodies by default
- runtime exposes an explicit delegated invocation path
- invoking a delegated skill launches the resolved package-local child-agent executor once #108 and #175 are available
- default delegated launch remains non-inheriting and explicit-handle-based

### PR DESCRIPTION
**Pull request by realmorrisliu**
## Summary
- replace delegated skill placeholder behavior with real parent-side delegated invocation stubs
- route `invoke_delegated_skill` through package-local child-agent launches
- keep delegated fallback behavior explicit for runtimes that intentionally disable nested delegation

## Testing
- cargo test -p alan-runtime
- cargo test -p alan cli::skills
- cargo test -p alan --test skills_cli_integration_test

Closes #183
Related to #181

### CODE REVIEW

File: crates/runtime/src/runtime/virtual_tools.rs [hunk 1]
```diff
@@ -262,88 +273,329 @@ where
...
+    });
+    emit(Event::ToolCallCompleted {
+        id: tool_call.id.clone(),
+        result_preview: preview,
+        audit: None,
+    })
+    .await;
+    state
+        .session
+        .record_tool_call(&tool_call.name, persisted_arguments, payload.clone(), false);
```
[the remark this answers is not in the document]
Comment (realmorrisliu) [reply crates/runtime/src/runtime/virtual_tools.rs]: Fixed in `66d0630`.

- `handle_invoke_delegated_skill()` now derives the persisted tool-call `success` bit from `DelegatedSkillResultStatus` instead of hard-coding `false`
- added a regression test that runs a successful delegated invocation with a real session recorder and asserts the rollout `ToolCallRecord` is persisted as `success: true`

I also kept the CI fix from `d9070b0` in the same branch, since that Ubuntu-only pre-cancelled launch race was what broke the latest PR checks.

File: crates/runtime/src/runtime/child_agents.rs [hunk 2]
```diff
@@ -236,104 +348,175 @@ impl ChildRuntimeController {
...
-                                status: ChildRuntimeStatus::Paused,
-                            });
-                        }
-                        alan_protocol::Event::Error {
-                            message,
-                            recoverable,
-                        } if !recoverable => {
-                            return Ok(ObservedChildTerminalEvent {
-                                output_text,
+                        _ = timeout_sleep.as_mut().as_pin_mut().unwrap(), if timeout_sleep.is_some() => {
```
[the remark this answers is not in the document]
Comment (realmorrisliu) [reply crates/runtime/src/runtime/child_agents.rs]: Fixed in `fa57ac3`.

The review was right: in `tokio::select!`, the branch expression is evaluated before the guard, so the earlier `timeout_sleep.as_mut().as_pin_mut().unwrap()` could still panic when delegated execution ran with no timeout.

I rewrote `wait_for_terminal_event_with_optional_cancel()` to split the four cases explicitly:
- timeout + cancel
- timeout only
- cancel only
- neither

That removes the panic-prone `unwrap()` entirely.

I also added a controller-level regression test covering `join_until_cancelled()` with `timeout = None`, so the no-timeout delegated path is now exercised directly. I rechecked `child_agents.rs` for the same select/guard pattern after the fix; this was the only remaining instance.
